src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -264,10 +270,70 @@ private int LineBreaksAfter(SyntaxToken currentToken, SyntaxToken nextToken)
             return 0;
         }
 
+        private static bool All<TNode>(SyntaxList<TNode> items, Func<TNode, bool> predicate)
+            where TNode : SyntaxNode
+        {
+            bool result = true;
+            for (int i = 0; i < items.Count; i++)
+            {
+                result &= predicate(items[i]);
+            }
+
+            return result;
+        }
+
+        private static bool IsAutoAccessorList(SyntaxToken token)
+            => token.Parent is AccessorListSyntax accessorList &&
+                All(accessorList.Accessors, a => a.Body == null);
```

Comment:
>a.Body == null [](start = 49, length = 14)

What about presence of ```ExpressionBody```? Do we have tests with expression-bodied accessors?

Reply:
I used the absence of Body in both accessors as the trigger to format the property inline instead of separate lines.
It is basically the same strategy used by VS and other "style cops" I had occasion to use.

Reply:
> I used the absence of Body in both accessors

Then the check has nothing to do with the "auto" aspect. Correct? Then name of the method shouldn't imply that it does.

---
In reply to: [531002735](https://github.com/dotnet/roslyn/pull/49495#discussion_r531002735) [](ancestors = 531002735)